internal/config/config.go

```diff
@@ -23,42 +25,195 @@ import (
 	"golang.org/x/exp/maps"
 )
 
-// TestsConfig represents a part of the dance configuration for tests.
-// (i.e. ferretdb/mongodb/common tests).
+const (
+	// RunnerTypeCommand indicates a command-line test runner.
+	RunnerTypeCommand RunnerType = "command"
+
+	// RunnerTypeGoTest indicates a Go test runner.
+	RunnerTypeGoTest RunnerType = "gotest"
+
+	// RunnerTypeJSTest indicates a JavaScript test runner.
+	RunnerTypeJSTest RunnerType = "jstest"
+
+	// RunnerTypeYCSB indicates a YCSB test runner.
+	RunnerTypeYCSB RunnerType = "ycsb"
+)
+
+// RunnerType represents the type of test runner used in the configuration.
+type RunnerType string
+
+// Stats represent the expected/actual amount of
+// failed, skipped and passed tests.
 //
-// It's used to store information about expected test results for a
-// specific database.
+//nolint:musttag // we don't care about annotations there
+type Stats struct {
+	UnexpectedRest int
+	UnexpectedPass int
+	UnexpectedFail int
+	UnexpectedSkip int
+	ExpectedPass   int
+	ExpectedFail   int
+	ExpectedSkip   int
+}
+
+// Config represents the configuration settings for the test execution.
 //
-// May contain prefixes; the longest prefix wins.
-type TestsConfig struct {
-	Default status
+//nolint:govet // we don't care about alignment there
+type Config struct {
+	Runner RunnerType
+	Dir    string
+	// Args contains additional arguments for the test runner.
+	Args    []string
+	Results Results
+}
+
+// Results stores the expected test results for different databases.
+type Results struct {
+	Common   *TestConfig
```

Comment:
Let's remove that field. The idea that `configload` will do all needed transformations, merges, etc, and then the rest of the code could work with a simple `Config` type by using `Config.FerretDB` or `Config.For("ferretdb")`

Reply:
Do you mean here